### PR TITLE
Forms Automated Test - LPS-133463 | master 

### PR DESCRIPTION
**Pull request by fmarcio**

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsFieldsets.testcase [hunk 1]
```diff
@@ -219,14 +219,67 @@ definition {
...
+
+		Form.editName(formName = "Form 1");
+
+		for (var element : list "Element Set 1,Element Set 1") {
+			Form.gotoAddElementSet(elementSetTitle = "${element}");
+		}
+
+		Form.save();
+
+		Alert.viewSuccessMessage();
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsFieldsets.testcase:261]: @fmarcio, this step is already executed inside the macro `Form.save`, therefore you can remove it. :wink:

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro [hunk 2]
```diff
@@ -794,6 +794,13 @@ definition {
 			value1 = "${fieldValue}");
 	}
 
+	macro viewFieldLabelByDataFieldName {
+		AssertElementPresent(
+			fieldLabel = "${fieldLabel}",
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:799]: @fmarcio, this is not using the pattern used by the other macros, can you please update according to the example below?
```
key_fieldLabel = "${fieldLabel}",
```

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro [hunk 3]
```diff
@@ -794,6 +794,13 @@ definition {
 			value1 = "${fieldValue}");
 	}
 
+	macro viewFieldLabelByDataFieldName {
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:797]: @fmarcio, do you really need to create this macro or the macro `viewFieldLabel` can help you here? Can you please double check?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsFieldsets.testcase (hunk 1 above)
Comment (fmarcio) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormsFieldsets.testcase]: > @fmarcio, this step is already executed inside the macro `Form.save`, therefore you can remove it.

@natocesarrego fixed! :-)

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro (hunk 2 above)
Comment (fmarcio) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro]: > @fmarcio, this is not using the pattern used by the other macros, can you please update according to the example below?
> 
> ```
> key_fieldLabel = "${fieldLabel}",
> ```

@natocesarrego fixed! :-)

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro (hunk 3 above)
Comment (fmarcio) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:981]: > @fmarcio, do you really need to create this macro or the macro `viewFieldLabel` can help you here? Can you please double check?

@natocesarrego Unfortunately, `viewFieldLabel` did not work on my test. The test adds field labels with the same name, and the macro you suggested does not filter according to position :-(

I tried some labels macros who uses index position but none of them worked either. So I had to create this one which filters according to the data-field-name :-)

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro (hunk 2 above)
Comment (natocesarrego) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro]: @fmarcio, the new macro's code is:
```
var key_fieldName = "${key_fieldName}";
var key_fieldLabel = "${key_fieldLabel}";
```
However, the expected pattern is:
```
var key_fieldName = "${fieldName}";
var key_fieldLabel = "${fieldLabel}";
```
Can you please fix?

Thanks.

Comment (fmarcio) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro]: > @fmarcio, the new macro's code is:
> 
> ```
> var key_fieldName = "${key_fieldName}";
> var key_fieldLabel = "${key_fieldLabel}";
> ```
> 
> However, the expected pattern is:
> 
> ```
> var key_fieldName = "${fieldName}";
> var key_fieldLabel = "${fieldLabel}";
> ```
> 
> Can you please fix?
> 
> Thanks.

@natocesarrego fixed! :-)

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**